include/pybind11/pybind11.h

```diff
@@ -403,7 +408,14 @@ class cpp_function : public function {
         strdup_guard guarded_strdup;
 
         /* Create copies of all referenced C-style strings */
-        rec->name = guarded_strdup(rec->name ? rec->name : "");
+        if (rec->name == nullptr) {
+            rec->name = guarded_strdup("");
+        } else if (std::strcmp(rec->name, "@__setstate__") == 0) {
+            // See google/pywrapcc#30094 for background.
+            rec->name = guarded_strdup(rec->name + 1);
```

Comment:
Consider typing the string `__setstate__` explicitly to decouple the user provided descriptive method name "spec",  and the python member name.

Reply:
Done.